cloudcomposerdiff/README.md

```diff
@@ -0,0 +1,184 @@
+# A python command line tool to diff two Cloud composer environments.
+
+## Running this code
+
+### Clone repository
+
+Clone the code from GitHub to your local machine
+
+### Install locally with pip
+
+```
+pip install cloudcomposerdiff
+```
+
+### Authorizing the application using gcloud
+
+Install [gcloud](https://cloud.google.com/sdk/gcloud)
+
+Select a user which has access to both composer environments to compare.
+
+Wit this user account, initialize gcloud:
+
+```
+gcloud init
+```
+
+Next, authorize the application to access cloud composer environments:
+
+```
+gcloud auth application-default login
+```
+
+### Use the appplication
+
+```
+cloudcomposerdiff \
+--env1_project_id <> \
+--env1_location <> \
+--env1_name <> \
+--env2_project_id <> \
+--env2_location <> \
+--env2_name <>
+```
+
+## Developing this application
+
+### Clone repository
+
+Clone the code from GitHub to your local machine
+
+### Getting the correct version of python with pyenv
+
+Install [pyenv](https://github.com/pyenv/pyenv).
+
+Use pyenv to install a suitable version of python, e.g. 3.10.0
+```
+$ pyenv install 3.10.0    
+```
+
+Use pyenv to switch to python 3.10.0
+```
+$ pyenv shell 3.10.0
+```
+
+### Create a virtual environment with venv module
+
+```
+python3 -m venv venv
+```
+
+activate the virtual environment
+```
+source venv/bin/activate
+```
+check that we are using the python of the virtual environment
+```
+(venv) $ which python
+.../cloudcomposerdiff/venv/bin/python
+```
+
+### Install the package in editable mode
+
+Install the package locally within the created virtual environment.
+
+Do so in "editable mode" so that the package will reflect code changes.
+
+Navigate to the directory containing pyproject.toml
+
+Install all the dependcies end-users would get when running pip install
+
+```
+(venv ) $ python3 -m pip install --editable .
+```
+Install additiona dev only dependencies that end-users do not get
+```
+python -m pip install --editable .[dev]
+```
+
+with the above done, running the package will include code changes. This can
+be used for testing during development.
+```
+cloudcomposerdiff \
+--env1_project_id <> \
+--env1_location <> \
+--env1_name <> \
+--env2_project_id <> \
+--env2_location <> \
+--env2_name <>
+```
```

Comment:
This command is included here without much else in the name of instruction. Should it be here? If so, add a quick note about why you'd run it and the expected output, and please refactor it to reflect the suggestion I made earlier that removes the brackets

Reply:
Good point ! 

I have added some more text here to explain what's going on.

As per my other comments, I have also added 3 GIFs to hopefully make the concept of the tool super clear !